src/adapter/src/coord/message_handler.rs

```diff
@@ -67,7 +67,44 @@ impl<S: Append + 'static> Coordinator<S> {
             Message::LinearizeReads(pending_read_txns) => {
                 self.message_linearize_reads(pending_read_txns).await;
             }
+            Message::StorageUsage => {
+                self.storage_usage_update().await;
+            }
+        }
+    }
+
+    #[tracing::instrument(level = "debug", skip_all)]
+    async fn storage_usage_update(&mut self) {
+        let object_id = None;
+        let shard_sizes = self.storageusageclient.shard_sizes().await;
```

Comment:
@pH14 - What is the scope of the shards that are returned here? I have been assuming that they were scoped to the specific Environment that environmentd is running in (aka, representing only the storage that this environment is using), but I'd like to confirm that

Reply:
we'll pull down all the storage usage by this customer _in this AWS region_

e.g. if they ran in us-east-1 and us-east-2 they would have separate usage numbers

Reply:
(this has to do with how we setup S3 buckets. we currently create a customer-persisted-data bucket per region, and then give each customer a prefix within that region. `shard_sizes()` trawls over the customer prefix within the regional bucket)

Reply:
Ok, so effectively, since we only allow an account to have one mz Region|Environment per AWS region, that will map 1:1 with a single customer environment. Perfect :ok_hand: